### PR TITLE
DOC: fix PR02 errors in docstring for pandas.core.groupby.DataFrameGroupBy.hist

### PR DESCRIPTION
**Pull request by jordan-d-murphy**
All PR02 Errors resolved in the following cases:

1. scripts/validate_docstrings.py --format=actions --errors=PR02 pandas.core.groupby.DataFrameGroupBy.hist

- [x] xref https://github.com/pandas-dev/pandas/issues/57111
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -2515,6 +2514,92 @@ def hist(
...
+        .. plot::
+            :context: close-figs
+
+            >>> data = {
+            ...     "length": [1.5, 0.5, 1.2, 0.9, 3],
+            ...     "width": [0.7, 0.2, 0.15, 0.2, 1.1],
+            ... }
+            >>> index = ["pig", "rabbit", "duck", "chicken", "horse"]
+            >>> df = pd.DataFrame(data, index=index)
+            >>> hist = df.hist(bins=3)
```
Comment (mroeschke) [inline pandas/core/groupby/generic.py:2601]: Could you make this a groupby example?

Comment (jordan-d-murphy) [reply pandas/core/groupby/generic.py:2701]: you bet, pushing a fix now!

**Comment by jordan-d-murphy** [comment]
@mroeschke  can you review my updated example for df.groupby().hist() when you get a chance?

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jordan-d-murphy

**Pull request merged**